File: frontend/src/types/stock.ts
export interface StockDto {
    id: number;
    itemId: number;
    shelfId: number;
    quantity: number;
    lastUpdated: string;
    item?: {
        id: number;
        name: string;
        price: number;
    };
    shelf?: {
        id: number;
        code: string;
    };
}

export interface CreateStockDto {
    itemId: number;
    shelfId: number;
    quantity: number;
}

export interface UpdateStockDto {
    itemId: number;
    shelfId: number;
    quantity: number;
} 

File: WarehouseManagementSystem.Api/Models/Stock.cs
using System;
using System.Collections.Generic;

namespace WarehouseManagementSystem.Api.Models
{
    public class Stock
    {
        public int Id { get; set; }
        public int ItemId { get; set; }
        public Item Item { get; set; } = null!;
        public int ShelfId { get; set; }
        public Shelf Shelf { get; set; } = null!;
        public int Quantity { get; set; }
        public DateTime LastUpdated { get; set; }
    }
} 